crc/services/file_service.py

```diff
@@ -436,3 +439,109 @@ def publish_to_github(file_ids):
                     branch=target_branch
                 )
                 return {'updated': True}
+
+    @staticmethod
+    def dmn_from_spreadsheet(ss_data):
```

Comment:
This is a really long and complex function.  You might think about pulling this out into it's file, and splitting it up into distinct methods.  It would make testing this easier, and long term maintainability would be improved.